packages/syft/src/syft/oblv/model.py

```diff
@@ -0,0 +1,441 @@
+        }
+        req = requests.post(
+            self.__conn_string + "/tensor/action?op={}".format(action),
+            data=body,
+            files={"file": file},
+        )
+        # req = requests.post(self.__conn_string + "/tensor/action?op={}".format(action)
+        # , data=body, files={"file": file},headers={"x-oblv-user-name": "vinal", "x-oblv-user-role": "user"})
+        if req.status_code == 401:
+            raise OblvUnAuthorizedError()
+        elif req.status_code == 400:
+            raise OblvEnclaveError(req.json()["detail"])
+        elif req.status_code == 422:
+            print(req.text)
+            # ToDo - Update here
+            return "Failed"
+        elif req.status_code != 200:
+            raise OblvEnclaveError(
+                "Request to publish dataset failed with status {}".format(
+                    req.status_code
+                )
+            )
+        else:
+            # Status code 200
+            # TODO - Remove this after oblv proxy is resolved
+            data = req.json()
+            if type(data) == dict and data.get("detail") is not None:
+                raise OblvEnclaveError(data["detail"])
+            return data
+        # #return req.json()
+
+    def request_publish(self, dataset_id, sigma=0.5):
+        if len(self.client) == 0:
+            raise Exception(
+                "No Domain Clients added. Set the propert *client* with the list of your domain logins"
+            )
+        if self.__conn_string is None:
+            raise Exception(
+                "proxy not running. Use the method connect_oblv_proxy to start the proxy."
+            )
+        elif self.__conn_string == "":
+            raise Exception(
+                "Either proxy not running or not initiated using syft. Run the method initiate_connection"
+                + "to initiate the proxy connection"
+            )
+        depl = self.oblv_client.deployment_info(self.deployment_id)
+        if depl.is_deleted:
+            raise Exception(
+                "User cannot connect to this deployment, as it is no longer available."
+            )
+        req = requests.post(
+            self.__conn_string + "/tensor/publish/request",
+            json={"dataset_id": dataset_id, "sigma": sigma},
+        )
+        if req.status_code == 401:
+            raise OblvUnAuthorizedError()
+        elif req.status_code == 400:
+            raise OblvEnclaveError(req.json()["detail"])
+        elif req.status_code == 422:
+            print(req.text)
+            # ToDo - Update here
+            return "Failed"
+        elif req.status_code != 200:
+            raise OblvEnclaveError(
+                "Request to publish dataset failed with status {}".format(
+                    req.status_code
+                )
+            )
+        else:
+            # Status code 200
+            # TODO - Remove this after oblv proxy is resolved
+
+            data = req.json()
```

Comment:
MAYBE: The same question regarding `json()` -- it should always return a dict, right?

Reply:
Ping @vinalb-oblivious